plugins/state_history_plugin/state_history_plugin.cpp

```diff
@@ -87,11 +87,7 @@ struct state_history_plugin_impl : std::enable_shared_from_this<state_history_pl
    using acceptor_type = std::variant<std::unique_ptr<tcp::acceptor>, std::unique_ptr<unixs::acceptor>>;
    std::set<acceptor_type>          acceptors;
 
-   std::thread                                                              thr;
-   boost::asio::io_context                                                  ctx;
-   boost::asio::io_context::strand                                          work_strand{ctx};
-   boost::asio::executor_work_guard<boost::asio::io_context::executor_type> work_guard =
-       boost::asio::make_work_guard(ctx);
+   named_thread_pool                thread_pool{"SHiP", 1}; // use of executor assumes only one thread
```

Comment:
Do we want to be able to configure the number of SHiP threads?

Reply:
If we increase from 1 then we need to create and use a strand for each connection. I'm open to doing that if desired. I have a nice manual test setup for testing high threading if that is something we want to do now. @spoonincode thoughts?

Reply:
I would say no. imo the code needs to be substantially cleaned up before adding any more complexity to it.